### PR TITLE
Introduce TxId and TxHash types

### PR DESCRIPTION
**Pull request by t-bast**
It was really footgunny to only use a `ByteVector32` here, as users could use a `TxHash` when a `TxId` was expected. Introducing explicit types ensures that functions can clearly specify which one they expect.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -171,7 +171,7 @@
         <dependency>
             <groupId>fr.acinq.bitcoin</groupId>
             <artifactId>bitcoin-kmp-jvm</artifactId>
-            <version>0.13.0</version>
+            <version>0.14.0-SNAPSHOT</version>
```
Comment (sstone) [inline pom.xml:174]: You can now update `bitcoin-kmp` to 0.14.0 and `secp256k1-kmp` to 0.11.0

Comment (t-bast) [reply pom.xml]: Done in https://github.com/ACINQ/bitcoin-lib/pull/76/commits/0097a4467a4a41063e65ef300728035b087938e7

Comment (t-bast) [reply pom.xml]: Ah I'm missing the secp update, hold on.

Comment (t-bast) [reply pom.xml]: Done!

**Review by sstone — APPROVED** [review]

**Pull request merged**